.gitignore

```diff
@@ -1,3 +1,4 @@
 *~
 __pycache__
 .DS_Store
+settings.json
```

Comment:
Hm, where did this change come from? What is it for?

Reply:
This is from my VSCode placing a settings.json in the working directory automatically. It also automatically added this to the gitignore file. I can remove the file and reference to it in the gitignore file if necessary.

Reply:
That's fine, I was just wondering.

But in that case, we should write it as `/settings.json`, otherwise *any* file in this repo named `settings.json` will be ignored, and that might be an actual footgun in the future :)

Reply:
Oh, ya I had no idea about that, I'll rebase and commit that change,